### PR TITLE
fix image loading

### LINKED ISSUE #2998: Gallery app missing image assets
To reproduce, load up `examples/material_gallery` and notice that the top is a solid color instead of showing some images.

Probably a result of switching to `--no-package-symlinks`.

/cc @devoncarew @vlidholt

### PR DESCRIPTION
**Pull request by devoncarew**
- better fix for image loading
- fix #2998

### CODE REVIEW

**Comment by devoncarew** [comment]
I'll open an issue to get the tooling side of this under test.

**Comment by devoncarew** [comment]
@vlidholt @krisgiesing

**Comment by vlidholt** [comment]
LGTM, but perhaps someone else should check this too

File: packages/flutter_tools/lib/src/flx.dart [hunk 1]
```diff
@@ -34,14 +34,33 @@ const String _kFontSetMaterial = 'material';
...
 
-  final String source;
   final String base;
-  final String key;
+
+  /// The entry to list in the generated asset manifest.
+  String get assetEntry => _assetEntry ?? relativePath;
+
+  /// Where the resource is on disk realtive to [base].
+  final String relativePath;
```
Comment (krisgiesing) [inline packages/flutter_tools/lib/src/flx.dart]: Just looking at the comments, it's not so clear to me what the difference is between base and source.  I'm starting to think we should do a pass over Asset and come up with somewhat better names and documentation for these fields.

Comment (devoncarew) [inline packages/flutter_tools/lib/src/flx.dart]: Yup, from the one use of `source`, it looks like it's an alternative to `relativePath` (or what had been called `key`). +1 to a refactor and re-alignment

Comment (Hixie) [inline packages/flutter_tools/lib/src/flx.dart]: file a bug to track this?

**Comment by Hixie** [comment]
It would be ideal if we could have a test that checks that image loading works.

**Comment by krisgiesing** [comment]
Devon just added https://github.com/flutter/flutter/issues/3005 to write a test for manifest creation (which was the particular problem here). The unit test I have for testing asset variant loading has the filesystem calls mocked out, so it didn't catch that part of the problem. Do you have recommendations for other tests we could write?

File: packages/flutter_tools/lib/src/flx.dart (hunk 1 above)
Comment (krisgiesing) [inline packages/flutter_tools/lib/src/flx.dart]: https://github.com/flutter/flutter/issues/3031

**Pull request merged**